src/trans.rs

```diff
@@ -314,12 +314,12 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
                     let a = self.trans_operand(a);
                     let b = self.trans_operand(b);
                     let op = match *op {
-                        BinOp::Add => BinaryenAdd(),
-                        BinOp::Sub => BinaryenSub(),
-                        BinOp::Mul => BinaryenMul(),
-                        BinOp::Div => BinaryenDivS(),
-                        BinOp::Eq => BinaryenEq(),
-                        BinOp::Ne => BinaryenNe(),
+                        BinOp::Add => BinaryenAddInt32(),
```

Comment:
eventually there should be a check on the type, so we can handle the other types (i64, f32, f64). meanwhile maybe we could just assert that the type is i32? although, the binaryen validator should show an error if we get something like this wrong, so i guess it's ok as is too.